src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/PlatformDetection.Unix.cs

```diff
@@ -330,6 +330,57 @@ private static class libc
 
             [DllImport("System.Globalization.Native", SetLastError = true)]
             public static extern int GlobalizationNative_GetICUVersion();
+
+            delegate void GetICUVersionDelegate(IntPtr ptr);
+
+            public unsafe static int GetICUVersion()
```

Comment:
It would be best to get rid of the use of ICU version in the test. The tests can simply allow both results as valid, and the ICU versions that the different results apply to can be captured in comment.

Reply:
If you really want to keep testing the ICU version (I do not think it is useful...), I would call he internal CoreLib method that returns the ICU version via private reflection. It will work same way on both CoreCLR and Mono, and be simpler overall.

Reply:
> It would be best to get rid of the use of ICU version in the test. The tests can simply allow both results as valid, and the ICU versions that the different results apply to can be captured in comment.

Some of the tests skip test data for some versions of ICU because there is ICU fixes for overflows and stuff like that. https://github.com/dotnet/runtime/blob/master/src/libraries/System.Globalization/tests/CompareInfo/CompareInfoTests.IsPrefix.cs#L81

You don't think is worth doing that?

> ICU version via private reflection.

That's a good idea.

Reply:
> You don't think is worth doing that?

What does that test do on old ICU versions? Crash or just return wrong result?

Reply:
Hmm there is only two usages of ICUVersion in the tests, one the one that I pointed above and this one: https://github.com/dotnet/runtime/blob/9999d4b86e0c7b0e7fbc04c24bad346397ef17ff/src/libraries/System.Globalization/tests/System/Globalization/RegionInfoTests.cs#L145 -- which we could make both expected values valid in the tests.

> What does that test do on old ICU versions? Crash or just return wrong result?

I don't know, I'll let @tarekgh answer that, he should know more.

Reply:
We don't crash but ICU return wrong results. I am still seeing have ICUVersion is very useful for us. Sometimes it helps in diagnosing some raised issues without the need for deep debugging. Another minor reason is when writing the behavior of different ICU version in the test helps us to know the history of the change. This is really useful in some cases when it is needed instead of going back and try to figure out what has changed. 

In short, I don't mind changing the tests but I am still seeing a value keeping the ICU version property in our tests.